commands/test/test-command.ts

```diff
@@ -156,19 +156,67 @@ export const sendQuestion = async (interaction: any) => {
 
 
         const bestMatch = await findMatchingUser(interaction.user.id, userResponses);
+        console.log(bestMatch);
+        console.log(bestMatch?.userId);
 
         if (bestMatch) {
-            interaction.user.send(`Dein bester Gesprächspartner ist: **@${bestMatch.username}**.`);
-            interaction.user.send("Als nächstes schreibst du deinem Partner, indem du auf seinen Namen auf dem Contraversum-Server klickst 👆 und ihm eine Nachricht sendest.");
-            interaction.user.send("Dies sind drei Fragen bei denen ihr euch unterscheidet:");
+            const guildId = process.env.GUILD_ID;
+            if (!guildId) {
+                console.error('GUILD_ID is not defined in .env');
+                return;
+            }
+            const guild: Guild | undefined = client.guilds.cache.get(guildId);
+            if (!guild) {
+                console.error('Guild not found');
+                return;
+            }
 
-            // Send the best match that they have been matched with the user
-            const bestMatchUser = await client.users.fetch(bestMatch.userId);
-            if (bestMatchUser) {
-                bestMatchUser.send(`Hey 👋, du wurdest mit: **@${interaction.user.username}** gematched.`);
-                bestMatchUser.send("Ihr habt euch bei diesen drei Fragen beispielsweise unterschieden:");
+            const member = guild.members.cache.get(interaction.user.id);
+            if (!member) {
+                console.error('Member not found');
+                return;
+            }
+            const bestId = bestMatch.userId
+            console.log(bestId);
+
+            const bestMember = await guild.members.fetch(bestId);
```

Comment:
instead of having this awkward logic where we get the best user.id of the user name and returning the name again. Just change the findMatchingUser function to verify if the user is member of the server. That would be cleaner and less error prone

Reply:
Line 179 and 180 are definitely useless, but i don't get your point with changing the findMatchingUser function.

Reply:
the findMatchingUser function returns the userName. What we are doing is getting the userName then getting the userId of this UserName, then verifying if the user is a member of the server and then getting the userName of the userId of the userName that we got from the findMatchingUser function. We can make this cleaner by verifying in the findMatchingUser function if the user is a member of the server and then returning the userObject then we can use the user.id to create the channel. Maybe I am missing something but it does not seem to be optimal like this